### PR TITLE
refactor: tackle corner cases and change variable names for better readability

### PR DESCRIPTION
**Pull request by khareyash05**
Signed-off-by: Yash <yash2010118@akgec.ac.in>

- Completed TODO for `serveStatic` 
- Added corner cases and error handling for deleting applications
- Renamed variables so they are easier to read and understand their functionality
- Added more error cases for applications

### CODE REVIEW

File: src/utils/responseTexts.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+	app: string
+): {
+	success: string;
+	error: string;
+	folderShouldntExist: string;
+	appShouldntExist: string;
+} => ({
+	success: 'Deploy Delete Succeed',
+	error: `Oops! It looks like the application (${app}) hasn't been deployed yet. Please deploy it before you delete it.`,
+	folderShouldntExist: `The folder shouldnt exist after deleting it.`,
```
Comment (Creatoon) [inline src/utils/responseTexts.ts:11]: remove these two newly added cases, as they are not required, as the response is for user not the developer.

Remove the logic for folderShouldntExist, appShouldntExist, as all the cases are handled properly.

Also Dont use multiple return statements.

File: src/api.ts [hunk 2]
```diff
@@ -65,15 +65,15 @@ export const callFnByName = (
...
+	childProcesses[app].send({
 		type: protocol.c,
 		fn: {
 			name,
 			args
 		}
 	});
 
-	cps[app].on('message', (data: childProcessResponse) => {
+	childProcesses[app].on('message', (data: childProcessResponse) => {
```
Comment (Creatoon) [inline src/api.ts:76]: ```childProcesses``` resembles to childProcess and its hard to understand, use something different or lets keep it as it is.

**Review by Creatoon — CHANGES_REQUESTED** [review]

File: src/utils/responseTexts.ts (hunk 1 above)
Comment (khareyash05) [reply src/utils/responseTexts.ts:11]: Hey ! Thanks for the review I really dont think the cases are being handled properly. for example when scaling it for future there may be cases when the system disconects or any other simultaneous processes may block the current proccesses. I think these cases will be need and thus multiple return statements are necessary

Comment (khareyash05) [reply src/utils/responseTexts.ts:11]: I see we shouldnt send it to the user but we need to handle these cases at the backend side IMHO

**Comment by viferga** [comment]
@khareyash05 for me it seems correct, I do not understand what @Creatoon said respect to the error messages. Can you update the conflicts to make it work again with the latest changes?

Once you do that I will merge your PR.

**Comment by khareyash05** [comment]
> @khareyash05 for me it seems correct, I do not understand what @Creatoon said respect to the error messages. Can you update the conflicts to make it work again with the latest changes?
> 
> Once you do that I will merge your PR.

Resolved the conflicts. Maybe we are unable to comprehend what @Creatoon wants to say. If there is something I might be missing. Let me know. Thanks @viferga

**Comment by viferga** [comment]
I passed the CI and after my change of considering all warnings as errors (of the linter), your CI execution failed:
https://github.com/metacall/faas/actions/runs/8294111642/job/22775896259?pr=38

Correct this linter issue and I will merge it.

**Comment by khareyash05** [comment]
> I passed the CI and after my change of considering all warnings as errors (of the linter), your CI execution failed: https://github.com/metacall/faas/actions/runs/8294111642/job/22775896259?pr=38
> 
> Correct this linter issue and I will merge it.

fixed. Kindly rerun the CI

**Comment by giarve** [comment]
> > I passed the CI and after my change of considering all warnings as errors (of the linter), your CI execution failed: https://github.com/metacall/faas/actions/runs/8294111642/job/22775896259?pr=38
> > Correct this linter issue and I will merge it.
> 
> fixed. Kindly rerun the CI

Done, sorry for the long wait

**Comment by viferga** [comment]
@khareyash05 seems to be still failing.

**Comment by viferga** [comment]
I am merging this and I will solve the issues by myself.

**Comment by khareyash05** [comment]
Thanks @viferga

**Pull request merged**